lib/datadog/tracing/contrib/karafka/patcher.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require_relative '../patcher'
+require_relative 'ext'
+require_relative 'distributed/propagation'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Karafka
+        # Patch to add tracing to Karafka::Messages::Messages
+        module MessagesPatch
+          def configuration
+            Datadog.configuration.tracing[:karafka]
+          end
+
+          def propagation
+            @propagation ||= Contrib::Karafka::Distributed::Propagation.new
+          end
+
+          # `each` is the most popular access point to Karafka messages, but not the only one
+          #  Other access patterns do not have a straightforward tracing avenue (e.g. `my_batch_operation messages.payloads`)
+          # @see https://github.com/karafka/karafka/blob/b06d1f7c17818e1605f80c2bb573454a33376b40/README.md?plain=1#L29-L35
+          def each(&block)
+            @messages_array.each do |message|
+              if configuration[:distributed_tracing]
+                trace_digest = Karafka.extract(message.metadata.headers)
+                Datadog::Tracing.continue_trace!(trace_digest) if trace_digest
+              end
+
+              Tracing.trace(Ext::SPAN_MESSAGE_CONSUME) do |span|
+                span.set_tag(Ext::TAG_OFFSET, message.metadata.offset)
+                span.set_tag(Ext::TAG_TOPIC, message.topic)
```

Comment:
We should add a few standard tags for any messaging system spans, as these enable richer product integration for messaging system observability:
```suggestion
                span.set_tag(Ext::TAG_OFFSET, message.metadata.offset)
                span.set_tag('messaging.destination', message.topic) # Please create a constant for this one, close to `Contrib::Ext::Messaging::TAG_SYSTEM`
                span.set_tag(Contrib::Ext::Messaging::TAG_SYSTEM, 'kafka')
```

Reply:
The same should be done for the span `Datadog::Tracing.trace(Ext::SPAN_WORKER_PROCESS) do |span|`.

Reply:
I think that's the last of it! Thank you so much, @nvh0412!

Reply:
Thanks @marcotc, let me know if the standard tags I’ve added are correct.